types/Cargo.toml

```diff
@@ -18,3 +18,7 @@ log = { version = "0.4", default-features = false }
 serde = { version = "1", default-features = false, features = ["derive"] }
 serde_json = { version = "1", default-features = false, features = ["alloc", "raw_value", "std"] }
 thiserror = "1.0"
+# TODO: a bit sad to have to pull in soketto to handle errors
+soketto = "0.4"
```

Comment:
hmm, can't you use [this](https://github.com/paritytech/jsonrpsee/blob/master/types/src/error.rs#L36) instead?

or do we want concrete error types for every error?